src/IterationTraits.php

```diff
@@ -225,16 +225,43 @@ public static function sortKeys(Iterator $iterator, $fn = null) {
      * Group all the the values into an array and return the result as a Sequence.
      *
      * @param Iterator $iterator
-     * @param callable  $fnToGroup
+     * @param callable $fnToGroup
+     * @param null $keys
      * @return Sequence
      */
-    public static function groupBy(Iterator $iterator, $fnToGroup) {
-        return self::wrapFunctionIntoSequenceOnDemand(function() use ($iterator, $fnToGroup) {
+    public static function groupBy(Iterator $iterator, $fnToGroup, $keys = null) {
```

Comment:
I think that could work as another possibility, but I would disagree that it should be the only way to init because that increases the complexity of the function. What do you think about adding another branch to `initKeysForGroupBy` to handle callables. Something like this maybe:

```php
if (is_callable($keys)) {
    return call_user_func($keys);
}
```